packages/ui5-library-writer/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import { join } from 'path';
+import { create as createStorage } from 'mem-fs';
+import type { Editor } from 'mem-fs-editor';
+import { create } from 'mem-fs-editor';
+import cloneDeep from 'lodash/cloneDeep';
+import type { UI5LibConfig } from './types';
+import { enableTypescript } from './options';
+import { mergeWithDefaults } from './data';
+
+/**
+ * Writes the template to the memfs editor instance.
+ *
+ * @param basePath - the base path
+ * @param ui5LibConfig - the Ui5 lib instance
+ * @param fs - the memfs editor instance
+ * @returns the updated memfs editor instance
+ */
+async function generate(basePath: string, ui5LibConfig: UI5LibConfig, fs?: Editor): Promise<Editor> {
+    const reuseLib = cloneDeep(ui5LibConfig);
+
+    if (!fs) {
+        fs = create(createStorage());
+    }
+
+    const libInput = mergeWithDefaults(ui5LibConfig);
+    basePath = join(basePath, libInput.libraryNamespace);
+
+    const tmplPath = join(__dirname, '..', 'templates');
+    const ignoreJSFiles = ['**/src/baselibrary/Example.js', '**/ExampleRenderer.js', '**/library.js'];
+    const ignore = reuseLib.typescript ? ignoreJSFiles : ['**/*.ts'];
+
+    fs.copyTpl(join(tmplPath, 'common', '**/*.*'), basePath, libInput, undefined, {
+        globOptions: { dot: true, ignore },
+        processDestinationPath: (filePath: string) => {
+            const separator = process.platform === 'win32' ? '\\' : '/';
```

Comment:
Do we need to use different separators per OS? We don't do this in any other writer, why is it required here?

Reply:
I had over complicated it by using underscores for the hidden files. I've removed this and there is no need for the different separators per OS.